cfpb/cfgov-refresh
# Integrate wagtail-sharing for sharing of draft content

This PR replaces #2664 with a refactored version of [wagtail-sharing](https://github.com/cfpb/wagtail-sharing/tree/shared-site) for improved sharing of Wagtail drafts.

## Additions

- Adds wagtail-sharing to `requirements/base.txt` and as an installed app.

## Removals

- Removes `before_serve_page` hook that would interfere with using e.g. content.localhost:8000 as a local sharing site.

## Changes

- Replace Wagtail fallback URL patterns with those from wagtail-sharing.

## Testing

- Install requirements, create a new local database, and run a local server:

 ```sh
 $ pip install -r requirements/local.txt
 $ ./drop-db.sh && create-mysql-db.sh && initial-data.sh
 $ ./runserver.sh
 ```

- Create a draft page under the root with title "testing" and verify that it is not visible at http://content.localhost:8000/testing/.
- In the Wagtail admin, go to Settings > Sharing Sites and create a new sharing site _for the default localhost:8000 site_ with domain `content.localhost` and port 8000.
- Verify that the draft page is now visible at http://content.localhost:8000/testing/.
- Publish the page, make some changes, and save a new draft.
- Verify that the published state on http://localhost:8000/testing/ and the draft state on http://content.localhost:8000/testing/ differ as expected.

## Screenshots

![image](https://cloud.githubusercontent.com/assets/654645/23187489/f7f76526-f858-11e6-93df-23c72b28261a.png)

## Todos

- ~~This PR has been opened against [a PR branch](https://github.com/cfpb/wagtail-sharing/pull/7) in wagtail-sharing so will need to be updated when that is merged in.~~ That PR has been merged, requirements have been updated in 48d28b4 to remove the branch reference.
- I need to incorporate feedback from @schaferjh and others about the banner text and exposing the sharing link in the admin.

## Checklist
